packages/scoped-elements/test/ScopedElementsMixin.test.js

```diff
@@ -256,8 +256,40 @@ describe('ScopedElementsMixin', () => {
       },
     );
     const el = await fixture(`<${tag}></${tag}>`);
+
     expect(el.shadowRoot.children[0]).to.be.an.instanceOf(FeatureC);
+    expect(getFromGlobalTagsCache(FeatureD)).to.be.undefined;
+  });
 
-    expect(getFromTagsCache(FeatureD)).to.be.undefined;
+  it('supports lazy loaded elements', async () => {
+    const tag = defineCE(
+      class extends ScopedElementsMixin(LitElement) {
+        static get scopedElements() {
+          return {
+            'feature-a': FeatureA,
+            'feature-b': none,
```

Comment:
what if we just default to this behavior for an unscoped tag? so there is no need to explicitly declare it, any tag we can't find we scope it and wait for when it's defined later

Reply:
The problem with that is that it would break compatibility with self defined components.

I thought to not scope them until the element is defined, changing the tag in the dom at that moment, but it would cause problems with `getScopedTagName` function if it's called before the element is defined 🤷‍♂️

Reply:
If you don't change the tag, it can accidentally take on a definition from a global element with that name. 

When using scoped elements, I think you should not inherit anything globally so breaking compatibility with self defined components is fine.  If a component doesn't ship it's class you will need to do `'feature-b': customElements.get('feature-b'),`.

Changing the tag name at a later point in time would probably be pretty trick to do?

Reply:
Yes, good point. So then I think we could scope all custom elements in template by default so then we could remove the lazy ones from `static get scopedElements` and even `getScopedTagName` would continue working as expected.